### PR TITLE
Implement GetKeyCodeMap & GetKeyCodeMap2

### LINKED ISSUE #3919: SmileBASIC 4 Not Booting/Softlocking
Once the application starts, the frame rate/speed skyrockets and program hangs on the loading screen. This line seems to be of interest:
`[ 875.059554] Debug <Critical> core\hle\service\service.cpp:ReportUnimplementedFunction:154: Assertion Failed! Unknown / unimplemented function 'GetKeyCodeMap2': port='set' cmd_buf={[0]=0x1000006, [1]=0x8, [2]=0x1000, [3]=0xB5C5110, [4]=0x0, [5]=0x0, [6]=0x0, [7]=0x0, [8]=0x49434653}`

I feel like it could have to do with the keyboard controls (KeyCodeMap), as the program allows a hardware keyboard for input.

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/4608938/yuzu_log.txt)

### PR DESCRIPTION
**Pull request by ogniK5377**
Closes #3919

### CODE REVIEW

File: src/core/hle/service/set/set.cpp [hunk 1]
```diff
@@ -31,6 +32,44 @@ constexpr std::array<LanguageCode, 17> available_language_codes = {{
...
+    German = 8,
+    Italian = 9,
+    Portuguese = 10,
+    Russian = 11,
+    Korean = 12,
+    ChineseSimplified = 13,
+    ChineseTraditional = 14,
+};
+
+static std::unordered_map<LanguageCode, KeyboardLayout> language_to_layout{{
```
Comment (lioncash) [inline src/core/hle/service/set/set.cpp:53]:
```suggestion
static constexpr std::array<std::pair<LanguageCode, KeyboardLayout>, 17> language_to_layout{{
```

and do the lookup with `std::find_if`. This way there's no runtime initializer used at runtime when the program starts, and no heap is used up for the lifetime of the program.

File: src/core/hle/service/set/set.cpp [hunk 2]
```diff
@@ -120,6 +159,35 @@ void SET::GetRegionCode(Kernel::HLERequestContext& ctx) {
     rb.Push(Settings::values.region_index);
 }
 
+void GetKeyCodeMapImpl(Kernel::HLERequestContext& ctx) {
+    const auto language_code = available_language_codes[Settings::values.language_index];
+    const auto key_code =
+        std::find_if(language_to_layout.begin(), language_to_layout.end(),
+                     [&](const auto& element) { return element.first == language_code; });
```
Comment (lioncash) [inline src/core/hle/service/set/set.cpp:166]:
```suggestion
                     [=](const auto& element) { return element.first == language_code; });
```

Given all that's being copied is a single u64

File: src/core/hle/service/set/set.cpp [hunk 3]
```diff
@@ -120,6 +159,35 @@ void SET::GetRegionCode(Kernel::HLERequestContext& ctx) {
     rb.Push(Settings::values.region_index);
 }
 
+void GetKeyCodeMapImpl(Kernel::HLERequestContext& ctx) {
+    const auto language_code = available_language_codes[Settings::values.language_index];
+    const auto key_code =
+        std::find_if(language_to_layout.begin(), language_to_layout.end(),
```
Comment (lioncash) [inline src/core/hle/service/set/set.cpp:165]:
```suggestion
        std::find_if(language_to_layout.cbegin(), language_to_layout.cend(),
```

File: src/core/hle/service/set/set.cpp [hunk 4]
```diff
@@ -120,6 +159,35 @@ void SET::GetRegionCode(Kernel::HLERequestContext& ctx) {
...
+                     [&](const auto& element) { return element.first == language_code; });
+    KeyboardLayout layout = KeyboardLayout::EnglishUs;
+    if (key_code == language_to_layout.end()) {
```
Comment (lioncash) [inline src/core/hle/service/set/set.cpp:168]:
```suggestion
    if (key_code == language_to_layout.cend()) {
```

**Comment by ChaseParate** [comment]
Using the code additions that @ogniK5377 with the latest version of Yuzu repository, I build the program, and this time it actually booted! There was a small popup screen, but then after I pressed A to dismiss it, the application crashed and I got this error.
[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/4795218/yuzu_log.txt)
Seems like another function hasn't been defined.

File: src/core/hle/service/set/set.cpp [hunk 5]
```diff
@@ -120,6 +159,35 @@ void SET::GetRegionCode(Kernel::HLERequestContext& ctx) {
...
 
+void GetKeyCodeMapImpl(Kernel::HLERequestContext& ctx) {
+    const auto language_code = available_language_codes[Settings::values.language_index];
+    const auto key_code =
+        std::find_if(language_to_layout.cbegin(), language_to_layout.cend(),
+                     [=](const auto& element) { return element.first == language_code; });
+    KeyboardLayout layout = KeyboardLayout::EnglishUs;
+    if (key_code == language_to_layout.cend()) {
+        LOG_ERROR(Service_SET,
+                  "Could not find keyboard layout for language index {} defaulting to english us",
```
Comment (VolcaEM) [inline src/core/hle/service/set/set.cpp:170]:
```suggestion
                  "Could not find keyboard layout for language index {}, defaulting to English-US",
```

**Pull request merged**